validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/Eth1DataManager.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.coordinator;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.common.primitives.UnsignedLong;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentSkipListMap;
+import tech.pegasys.artemis.datastructures.blocks.Eth1Data;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.pow.event.Eth1BlockEvent;
+import tech.pegasys.artemis.storage.events.SlotEvent;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class Eth1DataManager {
+
+  private final EventBus eventBus;
+  private final UnsignedLong genesisTime;
+
+  private NavigableMap<UnsignedLong, Eth1Data> eth1ChainCache = new ConcurrentSkipListMap<>();
+  private final UnsignedLong RANGE_CONSTANT =
+      Constants.SECONDS_PER_ETH1_BLOCK.times(Constants.ETH1_FOLLOW_DISTANCE);
+  private volatile UnsignedLong currentVotingPeriodStartTime;
+
+  static Eth1Data getEth1Data(Eth1BlockEvent eth1BlockEvent) {
```

Comment:
(optional) Naming nit:
```suggestion
  static Eth1Data createEth1Data(Eth1BlockEvent eth1BlockEvent) {
```

Reply:
done.